infra/gcp/namespaces/ensure-namespaces.sh

```diff
@@ -0,0 +1,159 @@
+    echo "       --kubeconfig-path | -k      path to kubeconfig if not in: $HOME/.kube/config [optional]" >&2
+    echo "example: $0 my-cluster --kubectl-path /custom/path/kubectl --kubeconfig-path /custom/path/kubeconfig"
+    echo >&2
+}
+
+function parse_args() {
+  # positional args
+  args=()
+
+  # named args
+  while [ "$1" != "" ]; do
+      case "$1" in
+          -p | --kubectl-path )         KUBECTL="$2";            shift;;
+          -k | --kubeconfig-path )      KUBECONFIG_PATH="$2";    shift;;
+          * )                           args+=("$1")             # if no match, add it to the positional args
+      esac
+      shift # move to next kv pair
+  done
+
+  # restore positional args
+  set -- "${args[@]}"
+
+  # set defaults
+  if [[ -z "${KUBECONFIG_PATH}" ]]; then
+      KUBECONFIG_PATH="$HOME/.kube/config";
+  fi
+
+  if [[ -z "${KUBECTL}" ]]; then
+    if ! [ -x "$(command -v kubectl)" ]; then
+      echo >&2 "kubectl not found in \$PATH and --kubectl-path flag is not set. Aborting.";
+      exit 1;
+    else
+      KUBECTL="kubectl"
+    fi
+  fi
+
+  if [ ${#args[@]} -lt 1 ]; then
+    usage
+    exit 1
+  fi
+
+  CLUSTER="${args[0]}"
+}
+
+function create_namespace() {
+  if [ $# != 1 ]; then
+    echo "create_namespace(name) requires 1 argument" >&2
+    return 1
+  fi
+  namespace="$1"
+
+  $KUBECTL create -c $CLUSTER --kubeconfig $KUBECONFIG_PATH namespace "$namespace"
+}
+
+function apply_role() {
+  if [ $# != 1 ]; then
+    echo "apply_role(name) requires 1 argument" >&2
+    return 1
+  fi
+  project_name="$1"
+
+  sed -e "s/{{namespace}}/$project_name/" "$ROLE_FILENAME" | $KUBECTL apply -c $CLUSTER --kubeconfig $KUBECONFIG_PATH -n "$namespace" -f -
+}
+
+function apply_role_binding() {
+  if [ $# != 1 ]; then
+    echo "apply_role_binding(name) requires 1 argument" >&2
+    return 1
+  fi
+  project_name="$1"
+  namespace="$project_name"
+
+  sed -e "s/{{namespace}}/$project_name/" "$ROLE_BINDING_FILENAME" | $KUBECTL apply -c $CLUSTER --kubeconfig $KUBECONFIG_PATH -n "$namespace" -f -
+}
+
+function ensure_only_one_proper_role_binding() {
+  if [ $# != 1 ]; then
+    echo "ensure_only_one_proper_role_binding(name) requires 1 argument" >&2
+    return 1
+  fi
+  project_name="$1"
+  namespace="$project_name"
+  role_bindings_count="$($KUBECTL get rolebindings -c $CLUSTER --kubeconfig $KUBECONFIG_PATH -o json -n "$namespace" | jq -r '.items | length')"
+
+  if [ "$role_bindings_count" -gt 1 ]; then
+    echo "Only one role binding per namespace is allowed (current: $role_bindings_count)"
+    exit 1
+  fi
+
+  role_binding_name="$($KUBECTL get rolebindings -c $CLUSTER --kubeconfig $KUBECONFIG_PATH -o json -n "$namespace" | jq -r '.items[0].metadata.name')"
+
+  if [ "$role_binding_name" != "namespace-user" ]; then
+    echo "Expected role binding: 'namespace-user', got: '$role_binding_name'"
+    exit 1
+  fi
+}
+
+parse_args "$@";
+
+#
+# Project names
+#
+
+# Example project
+EXAMPLE_PROJECT="example_project"
```

Comment:
no real need for variables here, you can put literals into the array

Reply:
Changed :-)